### DISCUSSION TITLE
Avoiding final shell inside the image

### DISCUSSION THREAD

**Discussion by mmadrigalCS** [gitbls/sdm#256] [Q&A] [answered]
Hi! 
I recently found sdm as part of looking for a good distribution process for my custom raspberry pi images and I really like it. However  I noted that once the sdm customize process finished it will open a terminal shell inside the image for further actions. My custom script wrapping up the sdm execution will not continue until I manually exit this shell.

Is there any way to avoid this to happen? I looked over the command list but I was not able to find anything related, maybe I'm missing some parameter?

**Comment by gitbls** [comment, marked as answer]
Does the `--batch` command switch, described on [this page](https://github.com/gitbls/sdm/blob/master/Docs/Command-Details.md) do what you want?

**Reply by mmadrigalCS** [reply]
Awesome! Thank you so much, it works as expected by exiting automatically the batch mode.

May I recommend to add this detail to the documentation? I didn't really get when or how this command was going to be executed, so it was easy to miss.

**Reply by gitbls** [reply]
Um, that page IS part of the documentation. Please enlighten me as to how the existing explanation of `--batch` should be changed to align with your perspective and documentation-reading capabilities. Thx

**Reply by mmadrigalCS** [reply]
Oh, not big deal there actually, if --batch command will only avoid the execution of the shell at the end of the process so the description may say so.  Instead of just say : "— Do not provide an interactive command prompt inside the nspawn container...", you could add something like " .. at the end of the process combined with --customize parameter ". or something similar, just to provide more context about how the command works.

The name of the command is also a little difficult to relate  with the functionality, however I know how hard to make a change like this may be. Maybe in the future you could use something like "--not-batch" which may be easier for people like me that just start looking around and understand how the tool works. 

Thanks a lot for the help!

**Reply by gitbls** [reply]
Thx for the perspective. Us _elders_ just know what `batch` means. I've updated the description a bit.
